### PR TITLE
feat: `recover_signer()` utility method in `ape.types.signature`

### PR DESCRIPTION
**Pull request by sabotagebeats**
### What I did
added recover_signer to ape.types.signatures

fixes: #1052

### How I did it
implemented recover_signer function in ape.types.signatures

### How to verify it
`ape test tests/functional/test_accounts.py::test_recover_signer`

### Checklist

- [x] All changes are completed
- [x] New test cases have been added
- [x] Documentation has been updated

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

File: src/ape/types/signatures.py [hunk 1]
```diff
@@ -33,6 +36,13 @@ class MessageSignature(_Signature):
     """
 
 
+def recover_signer(msg: SignableMessage, sig: MessageSignature) -> AddressType:
+    """
+    Recovers the signer of a message with the signature and the original message
```
Comment (unparalleled-js) [inline src/ape/types/signatures.py:41]: - [ ] Add docs for Args and return value

**Review by unparalleled-js — DISMISSED** [review]
Looks great! Missing args and return value docs however

**Review by unparalleled-js — APPROVED** [review]

**Review by unparalleled-js — DISMISSED** [review]

**Comment by unparalleled-js** [comment]
https://shipitsquirrel.github.io/images/ship%20it%20squirrel.png

**Review by sabotagebeats — COMMENTED** [review]
LGTM

**Review by unparalleled-js — APPROVED** [review]

**Pull request merged**